storage-node/src/cache/policy/mod.rs

```diff
@@ -50,13 +50,21 @@ impl DataStoreCacheValue for ParpulseDataStoreCacheValue {
 ///
 /// There are different cache policies for the data store cache, such as LRU, LRU-K, etc. See
 /// other files in this module for more details.
-pub trait DataStoreCache {
+pub trait DataStoreReplacer {
     /// Gets a value from the cache. Might has side effect on the cache (e.g.
     /// modifying some bookkeeping fields in the cache).
     fn get(&mut self, key: &ParpulseDataStoreCacheKey) -> Option<&ParpulseDataStoreCacheValue>;
 
     /// Puts a value into the cache.
-    fn put(&mut self, key: ParpulseDataStoreCacheKey, value: ParpulseDataStoreCacheValue) -> bool;
+    /// Returns success or failure and the evicted key if applicable.
+    fn put(
+        &mut self,
+        key: ParpulseDataStoreCacheKey,
+        value: ParpulseDataStoreCacheValue,
+    ) -> (bool, Option<Vec<ParpulseDataStoreCacheKey>>);
```

Comment:
If an Option is already returned, then I guess we do not need a boolean anymore.

Reply:
But do we need to distinguish `(true, None)` and `(false, None)`?